acq4/devices/DAQSonicator.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+
+from acq4.devices.DAQGeneric import DAQGeneric
+from acq4.devices.Sonicator import Sonicator
+from acq4.util.future import Future, future_wrap
+from neuroanalysis.stimuli import Sine
+from pyqtgraph.units import nF, V, µs
+
+
+class DAQSonicator(Sonicator):
+    """Sonicator controlled by a DAQ device.
+    Config
+    ------
+    capacitance : float
+        The capacitance of the piezoelectric transducer in Farads. This is used to calculate the voltage required to hit
+        a given frequency, as well as the limits of the voltage output.
+    max slew rate : float
+        The maximum safe slew rate of the piezoelectric transducer in V/μs. This is used to calculate the voltage
+        output (default 3.9 V/μs).
+    analog : dict
+        The config of the analog output channel. This controls the voltage output to the sonicator.
+    digital : dict
+        The config of the digital output channel. This controls the power to the sonicator.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._capacitance = self.config.get("capacitance", 65 * nF)
+        self._maxSlewRate = self.config.get("max slew rate", 3.9 * V / µs)
+        self._daq = DAQGeneric(
+            name=f"__sonicator{self.name()}DAQ",
+            config={
+                "channels": {
+                    "analog": self.config["analog"],
+                    "digital": self.config["digital"],
+                },
+            },
+        )
+
+    @future_wrap
+    def sonicate(self, frequency: float, duration: float, lock: bool = True, _future: Future = None):
+        if lock:
+            self.actionLock.acquire()
+        try:
+            # Calculate the voltage required to hit the desired frequency
+            voltage = self.calcVoltage(frequency)
+            wave = Sine(0, duration, frequency, voltage).eval(sample_rate=self._daq.sampleRate).data
+            numPts = len(wave)
+            cmd = {
+                "protocol": {"duration": duration},
+                self._daq.getDAQName("analog"): {
+                    "rate": 500_000,
+                    "numPts": numPts,
+                },
+                self._daq.name(): {
```

Comment:
No waveform needed; just set the holding to 1 during and 0 after
This will especially help with devices like yours that can't to time-sampled DIO